pallets/tx_fee/src/tests.rs

```diff
@@ -0,0 +1,245 @@
+use super::*;
+use crate as pallet_tx_fee;
+use codec::Encode;
+use smallvec::smallvec;
+use sp_core::H256;
+use sp_runtime::{
+	testing::{Header, TestXt},
+	traits::{BlakeTwo256, IdentityLookup},
+	Perbill,
+};
+use std::cell::RefCell;
+
+use frame_support::{
+	pallet_prelude::Get,
+	parameter_types,
+	traits::{ConstU32, ConstU64, Imbalance, OnUnbalanced},
+	weights::{
+		DispatchClass, GetDispatchInfo, WeightToFeeCoefficient, WeightToFeeCoefficients,
+		WeightToFeePolynomial,
+	},
+};
+use frame_system as system;
+use pallet_balances::Call as BalancesCall;
+use pallet_transaction_payment::{CurrencyAdapter, InclusionFee, Multiplier, NextFeeMultiplier};
+use sp_runtime::FixedPointNumber;
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Runtime>;
+type Block = frame_system::mocking::MockBlock<Runtime>;
+
+frame_support::construct_runtime!(
+	pub enum Runtime where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: system::{Pallet, Call, Config, Storage, Event<T>},
+		Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+		TransactionPayment: pallet_transaction_payment::{Pallet, Storage},
+		MrcTxFeePallet: pallet_tx_fee::{Pallet},
+	}
+);
+
+thread_local! {
+	static EXTRINSIC_BASE_WEIGHT: RefCell<u64> = RefCell::new(0);
+}
+
+pub struct BlockWeights;
+impl Get<frame_system::limits::BlockWeights> for BlockWeights {
+	fn get() -> frame_system::limits::BlockWeights {
+		frame_system::limits::BlockWeights::builder()
+			.base_block(0)
+			.for_class(DispatchClass::all(), |weights| {
+				weights.base_extrinsic = EXTRINSIC_BASE_WEIGHT.with(|v| *v.borrow()).into();
+			})
+			.for_class(DispatchClass::non_mandatory(), |weights| {
+				weights.max_total = 1024.into();
+			})
+			.build_or_panic()
+	}
+}
+
+parameter_types! {
+	pub static WeightToFee: u64 = 1;
+	pub static TransactionByteFee: u64 = 1;
+	pub static OperationalFeeMultiplier: u8 = 5;
+}
```

Comment:
@shannonwells  @aramikm  with respect to question below, it was a random find for me that this WeightToFree maps to
WEIGHT_TO_FEE, and same for  TRANSACTION_BYTE_FEE

Due to them being static types. I haven't figured why it happens but if i define them again with EXTRINSIC_BASE_WEIGHT compiler complains